internal/support/logging/criteria.go

```diff
@@ -80,3 +82,47 @@ func (criteria matchCriteria) match(le models.LogEntry) bool {
 
 	return true
 }
+
+func createConditions(conditions []bson.M, field string, elements []string) []bson.M {
+	keyCond := []bson.M{}
+	for _, value := range elements {
+		keyCond = append(keyCond, bson.M{field: value})
+	}
+
+	return append(conditions, bson.M{"$or": keyCond})
+}
+
+func (criteria matchCriteria) CreateQuery() map[string]interface{} {
```

Comment:
There shouldn't be mongo code in support.
I don't understand why the type LogMatcher is needed, maybe circular dependencies? If it is the case, moving the matchCriteria to internal/support/logging/criteria could be cleaner to me.

Reply:
I created the db.LogMatcher interface in order to move the existing "persistence" implementation (which has explicit mongo code) from support-logging to internal/pkg/db/mongo. Here's the existing code, so I don't understand why you say "There shouldn't be mongo code in support" because currently there is:
https://github.com/edgexfoundry/edgex-go/blob/140f76327c59478a9a18c932c500ece70163137a/internal/support/logging/mongo.go#L66

Perhaps you're trying to say we should have the goal of removing the mongo code from support-logging.

The LogMatcher implementation is the matchCriteria type supplied by support-logging which uses the previously imported mongo types in order to construct queries. Further, the matchCriteria type is populated from the support-logging REST controller:
https://github.com/edgexfoundry/edgex-go/blob/140f76327c59478a9a18c932c500ece70163137a/internal/support/logging/server.go#L60

So I was attempting to retain access to the matchCriteria type for the controller, while abstracting away the implementation for how the mongo query is assembled from the db layer.

Another possible approach might be to instead move the CreateQuery implementation into the internal/pkg/db/mongo, but then the question will be where does the MatchCriteria type get defined? It can't be in support-logging because that will cause a circular reference, so it will have to be in /pkg/models. But that's unfortunate because it isn't an externally usable model. Would it make sense to define it right in internal/pkg/db/types? It shouldn't be in the mongo package because the same type can be used for other databases...?

Reply:
Perhaps you're trying to say we should have the goal of removing the mongo code from support-logging.
Yes, that is my comment. 
What about moving it to internal/support/logging/criteria/match.go?